packages/material-ui-based-components/src/components/AutoComplete.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useCallback } from 'react';
+import { withRuleEngine } from '../shared/withRuleEngine';
+import { PROPS } from '../utils/types';
+import { Autocomplete, FormControl, FormHelperText, TextField } from '@mui/material';
+
+const AutoCompleteComponent = (props: PROPS) => {
+    const {
+        label, required, enumNames, enum: enums, enabled, id,
+        isError, description, errorMessage, name, value
+    } = props;
+
+    const changeHandler = useCallback((event: any, newValue: any) => {
+        console.log(event, newValue);
+        props.dispatchChange(newValue !== null ? newValue.value : undefined);
+    }, [props.dispatchChange]);
+
+    const blurHandler = useCallback((event: any) => {
+        props.dispatchBlur(event.target.value);
+    }, [props.dispatchBlur]);
+
+    const focusHandler = useCallback((event: any) => {
+        props.dispatchFocus(event.target.value);
+    }, [props.dispatchFocus]);
+
+    let options = [{ value: enums ? enums[0] : '', label: enumNames ? enumNames[0] : enums ? enums[0] : '' }];
+
+    if (enums) {
+        for (let i = 1; i < enums.length; i++) {
+            let ob = { value: enums[i], label: enumNames ? enumNames[i] : enums[i] };
+            options = [...options, ob];
+        }
+    }
+
+    return (
+        <FormControl
+            variant="outlined"
```

Comment:
variant should be dynamic. It will come from the props.properties['afs:layout'].variant and default value of this outlined

Reply:
Implemented dynamic variant